contracts/modules/TransferManager/CTM/CountTransferManager.sol

```diff
@@ -53,20 +53,33 @@ contract CountTransferManager is CountTransferManagerStorage, TransferManager {
         view
         returns(Result, bytes32)
     {
-        return _verifyTransfer(_from, _to, _amount);
+        uint256 holderCount = securityToken.holderCount();
+        if (_amount != 0 && _from != _to) {
+            // Check whether receiver is a new token holder
+            if (_to != address(0) && securityToken.balanceOf(_to) == 0) {
+                holderCount++;
+            }
+            // Check whether sender is moving all of their tokens
+            if (_amount == securityToken.balanceOf(_from)) {
+                holderCount--;
```

Comment:
guard `holderCount` from over/underflows as it can be zero. See existing implementation in `TokenLib.adjustInvestorCount`.

Reply:
@tintinweb Since `holderCount` can only be incremented/decremented by one at a time, it can not be overflowed.

`holderCount` cannot be underflowed either since `holderCount` must be greater than 1 if `_amount != 0 && _from != _to` (checked on L57).

Earlier, we used SafeMath everywhere (including `TokenLib.adjustInvestorCount`) but lately, I have using SafeMath only where it is needed (for optimization). 

Do you have a scenario in mind where holderCount can be underflowed or overflowed?

Reply:
thanks @maxsam4, agree the upper bound is unrealistic (especially when using ctm :)) and `holderCount==0` with `amount !=0` is mutually exclusive 👍